### DISCUSSION TITLE
How do I generate Node source maps for my dev server?

### DISCUSSION THREAD

**Discussion by ersinakinci** [remix-run/remix#9781] [Q&A]
I'm running Remix with Hono and Vite. When I have a server-side error in dev, I get stack traces without source maps.

I saw [this answer](https://github.com/remix-run/remix/discussions/8168#discussioncomment-7710245) from @kiliman to another question about source maps, which inspired me to debug by using `source-map-support` with the `retrieveSourceMap` hook that he suggested. I modified it to simply tell me where the library was attempting to load source maps from:

```js
sourceMapSupport.install({
  retrieveSourceMap: function (source) {
    console.log(source);
    ...
  },
});
```

I see that it's attempting to load source maps from various locations on my file system, from within my project directory. However, I don't see any `.map` files on the file system.

I know that Vite's dev server generates source maps for the browser bundle and makes these available through http. Those maps load fine in Chrome. But how do I get the maps for my Node environment?

(I'm surprised how difficult it is to find information on how these maps are generated by Vite, if at all, so this may not be the right place to ask. I'd appreciate anyone pointing me in the right direction.)

**Comment by fatso83** [comment]
Did you ever find out?

**Comment by kiliman** [comment]
`remix vite:build -sourcemapClient --sourcemapServer`

https://remix.run/docs/en/main/other-api/dev#remix-vitebuild

**Reply by fatso83** [reply]
Thanks, but that is for production. For development we just installed `source-map-support`, as mentioned in the docs.
